Infra: switch to linuxdeploy

SlySven opened the pull request:
#### Brief overview of PR changes/additions
Revises the Linux CI/CB process to use [`linuxdeploy`](https://github.com/linuxdeploy/linuxdeploy) instead of `linuxdeployqt`.

#### Motivation for adding to Mudlet
`linuxdeployqt` won't work next week as GitHub are removing the Ubuntu 20.04 runner and that tool will not work with the later `22.04` one until the maintainer updates it to do so (the CRT is too new).

#### Other info (issues closed, discussion etc)
The maintainer of linuxdeployqt is refusing to updating to Ubuntu 22.04 until it officially reaches EOL (in a month or so) but GitHub will not provide Ubuntu 20.04 from 2025/04/15 and has been "browning it out intermittently for the past few months.

This also switches the Linux run that is used for deployment to Ubuntu 22.04 it remains to be seen whether this is still usable for the oldest versions of the Linux OS which we provide AppImages for.

Some variables in the `bash` scripts used for Linux CI that were not have been made SCREAMING_SNAKE_CASE so that they can be picked out better.

Some issues that [ShellCheck](https://www.shellcheck.net/) highlighted have been "fixed" as well.

THIS IS A DRAFT CURRENTLY AS IT USES A PR FROM THE **INSTALLERS** REPO. It can not be merged until that PR is debugged and merged into that repository.

SlySven committed:
Fix: switch to linuxdeploy

#### Brief overview of PR changes/additions
Revises the Linux CI/CB process to use [`linuxdeploy`](https://github.com/linuxdeploy/linuxdeploy)
instead of `linuxdeployqt`.

#### Motivation for adding to Mudlet
`linuxdeployqt` won't work next week as GitHub are removing the Ubuntu
20.04 runner and that tool will not work with the later `22.04` one until
the maintainer updates it to do so (the CRT is too new).

#### Other info (issues closed, discussion etc)
The maintainer of linuxdeployqt is refusing to updating to Ubuntu 22.04
until it officially reaches EOL (in a month or so) but GitHub will not
provide Ubuntu 20.04 from 2025/04/15 and has been "browning it out
intermittently for the past few months.

This also switches the Linux run that is used for deployment to Ubuntu
22.04 it remains to be seen whether this is still usable for the oldest
versions of the Linux OS which we provide AppImages for.

Some variables in the `bash` scripts used for Linux CI that were not have
been made SCREAMING_SNAKE_CASE so that they can be picked out better.

Some issues that [ShellCheck](https://www.shellcheck.net/) highlighted have been "fixed" as well.

THIS IS A DRAFT CURRENTLY AS IT USES A PR FROM THE **INSTALLERS** REPO.
It can not be merged until that PR is debugged and merged into that
repository.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: switch to GCC 12 and Qt 6.9.0 to try and force some cache regeneration

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: another try to get Qt to install

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert: undo changes to build-mudlet.yml

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: try a fix that worked for others

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: fix broken yaml in previous commit

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: another try to fix yaml

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: another try

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: force the installation of an older python wheel package

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: upstream seem to have fixed jurple/install-qt-action

So switch to later GCC as before AND later Ubuntu 22.04.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: use my repo as that is where the PR code resides

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Fix_switchTo_linuxdepley

SlySven committed:
Tweak: switch a compiler version to force a GHA rebuild

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: ensure QMAKE variable is set before running installer make-installer.sh

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Debug: find out where the built mudlet executable ends up

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: insert missing double-quote

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: update to remove debug check & rebuild after installer repo. update

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert: undo a seemingly trivial change that actually broke build-mudlet.yml

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: add new directories to consider for AppImages from linuxdeploy

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: make the new search location conditional on being Linux OS

Also try and cause a new build to be done as GH Actions are currently
suffering problems according to status report.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: correct previously added path to look for lua C modules

Also add conditional qDebug() messages for the loading of translations.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: actually include the translation debug code this time

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Add: get dictionaries from a new location for the linuxdeploy installer case

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Fix_switchTo_linuxdepley

SlySven committed:
Merge branch 'development' into Fix_switchTo_linuxdepley

Conflicts resolved in:
* .github/workflows/build-mudlet.yml

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: update some components used in Linux builds to reflect later runner

Also trigger a rebuild to use an update to the installer repo branch in
use so that it includes code to support older CRTs when the AppImage is
used.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: make a trivial change to a comment to force a rebuild

So that it pulls in a change to the installer repository...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: reset gcc version used in later runner to be same as previous one

Using a later one might be the reason I couldn't get the AppImage to run
on Devuan 4.0 "Chimaera" even with using the `checkrt` plugin for
`linuxdeploy`.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

ZookaOnGit commented:
Debian 12.  Geyser seems to be acting funny.

FlexGUI before;
![image](https://github.com/user-attachments/assets/da0bb03d-f694-45df-a498-221cf8f80e16)

After;
![image](https://github.com/user-attachments/assets/06d00779-064c-4be5-9e35-930316ec1532)

My other Geyser profiles were affected too, mostly missing windows and decoration.  Then stumbled across this error about AdjustableContainers.

![image](https://github.com/user-attachments/assets/a3d51231-0d8a-4d11-b40b-3a823e7aa3f4)

Reproducable with;
```lua testCon = testCon or Adjustable.Container:new({name="testContainer"})```

No icons in right-click menu, can't manipulate window.

![image](https://github.com/user-attachments/assets/173a1623-5787-4bd4-88f0-12cec9c09067)

SlySven commented:
@ZookaOnGit thanks for that error report - that is the sort of detail this needs. Which locale have you set your GUI to - is it the (default) en_US - `English (American)` one - if so can you try changing to the `English (British)` one and see if the error is the same?


The reason for this is that I note that if I expand the contents of the AppImage (with `./Mudlet-4.19.1-testing-pr7806-d8e20675.AppImage --appimage-extract` on the command-line) the Lua "translations" for the other locales get expanded to `./squashfs-root/usr/share/applications/mudlet/lua/translations/` but not the one for the American locale.
![image](https://github.com/user-attachments/assets/0a71fb42-1cc9-4ebf-8169-103604e3d76d)

A quick fix to confirm that this is the problem would be to copy the `./translations/lua/mudlet-lua.json` file from the current source code and put it into that "expanded" `./squashfs-root/usr/share/applications/mudlet/lua/translations/` before trying to run the application via the `./squashfs-root/AppRun` shell-script. If you confirm that this is the detail I'll try and fix things so it gets copied to the right place...

It seems we hit the same issue in the past with Windows builds (see #7242!)

SlySven committed:
Revise: report the version of lrelease being used

This might be important if the Qt5 and Qt6 ones are different!

This is actually needed to prompt a rebuild with a revised PR in the
installer repository.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: try and get jurple/install-qt-action working again

Also report the lrelease tool version when it is run by CMake - unlike the
prior commit that only did it for QMake.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
Hey @SlySven, going to close this one out since it's been a while without movement. Your work isn't lost though - the code's still in your fork if you ever want to pick it back up. Thanks for the effort on this!
